components/RichText/RichText.tsx

```diff
@@ -81,21 +81,25 @@ const getRenderOptions = (links?: any) => {
                 node: Block | Inline,
                 _: React.ReactNode,
             ) => {
-                if (links) {
-                    const entryMap = new Map();
-                    for (const entry of links.entries.block) {
-                        entryMap.set(entry.sys.id, entry);
-                    }
+                try {
```

Comment:
I this seems fine, and you're right to be confident this will work. 

Ideally you'd catch the specific error as this code adds technical dept (lacks context).
So something like the below would be better, however extra comments would also adequate.

```
for (const entry of links.entries.block) {
    if (entry.sys === undefined) {
        raise sentry warning
        return <p>Embedded item not yet published.</p>
     }
    entryMap.set(entry.sys.id, entry);
}
```